rgf/sklearn.py

```diff
@@ -61,6 +67,113 @@ def sigmoid(x):
     """
     return 1. / (1.+ np.exp(-x))
 
+def _validate_params(max_leaf,
+                    test_interval,
+                    algorithm,
+                    loss,
+                    reg_depth,
+                    l2,
+                    sl2,
+                    normalize,
+                    min_samples_leaf,
+                    n_iter,
+                    n_tree_search,
+                    opt_interval,
+                    learning_rate,
+                    verbose,
+                    prefix,
+                    inc_prefix,
+                    clean,
+                    calc_prob="Sigmoid"):
+    if not isinstance(max_leaf, (numbers.Integral, np.integer)):
+        raise ValueError("max_leaf must be an integer, got {0}.".format(type(max_leaf)))
+    elif max_leaf <= 0:
+        raise ValueError("max_leaf must be greater than 0 but was %r." % max_leaf)
+        
+    if not isinstance(test_interval, (numbers.Integral, np.integer)):
+        raise ValueError("test_interval must be an integer, got {0}.".format(type(test_interval)))
+    elif test_interval <= 0:
+        raise ValueError("test_interval must be greater than 0 but was %r." % test_interval)
+        
+    if not isinstance(algorithm, six.string_types):
+        raise ValueError("algorithm must be a string, got {0}.".format(type(algorithm)))
+    elif not algorithm in _ALGORITHMS:
+        raise ValueError("algorithm must be 'RGF' or 'RGF_Opt' or 'RGF_Sib' but was %r." % algorithm)
+        
+    if not isinstance(loss, six.string_types):
+        raise ValueError("loss must be a string, got {0}.".format(type(loss)))
+    elif not loss in _LOSSES:
+        raise ValueError("loss must be 'LS' or 'Expo' or 'Log' but was %r." % loss)
+        
+    if not isinstance(reg_depth, (numbers.Integral, np.integer, _FLOATS)):
+        raise ValueError("test_interval must be an integer or float, got {0}.".format(type(reg_depth)))
+    elif reg_depth < 1:
+        raise ValueError("reg_depth must be no smaller than 1.0 but was %r." % reg_depth)
+        
+    if not isinstance(l2, _FLOATS):
+        raise ValueError("l2 must be a float, got {0}.".format(type(l2)))
+    elif l2 < 0:
+        raise ValueError("l2 must be no smaller than 0.0 but was %r." % l2)
+        
+    if not isinstance(sl2, (type(None), _FLOATS)):
+        raise ValueError("sl2 must be a float or None, got {0}.".format(type(sl2)))
+    elif sl2 is not None and sl2 < 0:
+        raise ValueError("sl2 must be no smaller than 0.0 but was %r." % sl2)
+        
+    if not isinstance(normalize, bool):
+        raise ValueError("normalize must be a boolean, got {0}.".format(type(normalize)))
+        
+    err_desc = "min_samples_leaf must be at least 1 or in (0, 0.5], got %r." % min_samples_leaf
+    if isinstance(min_samples_leaf, (numbers.Integral, np.integer)):
+        if not 1 <= min_samples_leaf:
```

Comment:
I found your recommendation more readable. Maybe we should nevertheless change it?